check-or-enforce-order.py

```
@@ -1,27 +1,34 @@
-import frontmatter
+import frontmatter as fm
 from pathlib import Path, PosixPath
 import sys
 
 # path here is intended to include only posts from a single language
+# _posts/r, _posts/plotly_js, _posts/python-v3, _posts/python in 'documentation'
+# build/html in 'plotly.py-docs'
 try:
     file_path = str(sys.argv[1])
 except:
     raise Exception("You need to specify a path!")
 
-# check to see if enforce flag was given
+# check to see if enforce flag was given at command line
 enforce = False
 if len(sys.argv) == 3:
     if sys.argv[2] == 'enforce':
         enforce = True
 
-# post families with these strings as "display_as" front-matter will be checked
 categories = ["file_settings", "basic", "financial", "statistical", "scientific", "maps", "3d_charts", "multiple_axes"]
 
-paths = []
-for suffix in ["md", "html"]:
-    paths += [x for x in Path(file_path).glob("**/*."+suffix)]
+def get_paths():
+    paths = []
+    for suffix in ["md", "html"]:
+        paths += [x for x in Path(file_path).glob("**/*."+suffix)]
+    return paths
 
-def get_meta(post):
+def get_post(path):
+    if ".ipynb_checkpoints" not in str(path):
```

It's an anti-pattern to conditionally return something from a function... what happens "else" here? Nothing is returned? won't that cause an exception downstream? you should probably raise an exception here and wrap your call below in a try/except that does `continue` in the `except` or something